editor/src/clj/editor/model_scene.clj

```diff
@@ -172,46 +172,70 @@
                 (some some? texcoord-datas) (assoc :texcoord-datas texcoord-datas)))
       (error-values/error-fatal "Failed to produce vertex buffers from mesh set. The scene might contain invalid data."))))
 
-(defn mesh->vb! [^VertexBuffer vbuf ^Matrix4d world-transform ^Matrix4d normal-transform vertex-attribute-bytes mesh-renderable-data]
+(defn- matrix4->bytes [^Matrix4d mtx]
+  (let [matrix-array (math/vecmath->clj (doto (Matrix4d. mtx) (.transpose)))
+        byte-array (byte-array (* 4 16))
+        byte-buffer (vtx/wrap-buf byte-array)]
+    (vtx/buf-push! byte-buffer :float false matrix-array)
+    byte-array))
```

Comment:
This function currently doesn't take the data type into account - does it make sense to convert a world matrix to a "byte matrix"?

Reply:
I think we need to do an actual type conversion from mat4 into mat3 (or mat2) here.